sqlglot/schema.py

```diff
@@ -209,38 +237,55 @@ def add_table(
 
         parts = self.table_parts(normalized_table)
 
-        nested_set(
-            self.mapping,
-            tuple(reversed(parts)),
-            normalized_column_mapping,
-        )
+        nested_set(self.mapping, tuple(reversed(parts)), normalized_column_mapping)
         new_trie([parts], self.mapping_trie)
 
-    def column_names(self, table: exp.Table | str, only_visible: bool = False) -> t.List[str]:
-        table_ = self._normalize_table(self._ensure_table(table))
-        schema = self.find(table_)
+    def column_names(
+        self,
+        table: exp.Table | str,
+        only_visible: bool = False,
+        dialect: DialectType = None,
+    ) -> t.List[str]:
+        dialect = dialect or self.dialect
```

Comment:
you shouldn't have to do this in every place, normalize_table and ensure_table should handle it right

Reply:
good point thanks